### PR TITLE
bugfix for issue #2419

### PR DESCRIPTION
**Pull request by rebert**
@loicgasser : 
- this fixed the bug #2419 
- no more layer specific ch.swisstopo.zeitreihen #1056  

[try](https://mf-geoadmin3.dev.bgdi.ch/src/?api_url=%2F%2Fmf-chsdi3.int.bgdi.ch%2Fltrea_bugfix_zeitreihe&topic=ech&lang=de&bgLayer=ch.swisstopo.pixelkarte-grau&layers=ch.swisstopo.zeitreihen,ch.bfs.gebaeude_wohnungs_register,ch.bav.haltestellen-oev,ch.astra.wanderland&layers_visibility=true,false,false,false&layers_timestamp=18641231,,,&X=199858.28&Y=599908.66&zoom=6&catalogNodes=687,720)

### CODE REVIEW

**Comment by rebert** [comment]
jenkins test

**Comment by rebert** [comment]
@ltclm : I have adapted the code after Jenkins feedback

File: chsdi/models/__init__.py [hunk 1]
```diff
@@ -79,7 +79,7 @@ def models_from_bodid(bodId, scale=None, resolution=None):
     if models:
         if scale is not None:
             models = [m for m in models if scale < max_scale(m) and scale >= min_scale(m)]
-        elif resolution is not None:
+        if resolution is not None:
```
Comment (loicgasser) [inline chsdi/models/__init__.py]: This doesn't work. What if scale and resolution are defined (which is the case most of the time)?
You would override the first result.

File: chsdi/views/features.py [hunk 2]
```diff
@@ -205,12 +205,13 @@ def _identify(request):
...
     isScaleDependent = hasBuffer(params.imageDisplay, params.mapExtent, params.tolerance)
     scale = getScale(params.imageDisplay, params.mapExtent) if isScaleDependent else None
+    resolution = getResolution(params.imageDisplay, params.mapExtent) if isScaleDependent else None
     if params.layers == 'all':
         model = get_bod_model(params.lang)
         query = params.request.db.query(model)
         for layer in get_layers_metadata_for_params(params, query, model):
             layerBodId = layer['layerBodId']
-            models = models_from_bodid(layerBodId, scale=scale)
+            models = models_from_bodid(layerBodId, scale=scale, resolution=resolution)
```
Comment (loicgasser) [inline chsdi/views/features.py]: I am in favour of simply not using min/max resolution in identify requests.

**Comment by loicgasser** [comment]
As a side note bgdi_order refers to the type of maps:
```sql
                        CASE
                            WHEN historische_karten.produkt = 'lk25'::text THEN 1
                            WHEN historische_karten.produkt = 'lk50'::text THEN 2
                            WHEN historische_karten.produkt = 'lk100'::text THEN 3
                            WHEN historische_karten.produkt = 'ta25'::text THEN 4
                            WHEN historische_karten.produkt = 'ta50'::text THEN 5
                            WHEN historische_karten.produkt = 'tk100'::text THEN 6
                            ELSE 99
                        END AS bgdi_order
```

**Comment by loicgasser** [comment]
I will close this one as I have an alternative solution, thanks a lot for that @rebert

**Pull request closed without merging**